Backpack-SwiftUI/ImageGalleryGrid/Classes/BPKImageGalleryGrid.swift

```diff
@@ -0,0 +1,230 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+
+struct ImageGalleryGrid<ImageView: View>: ViewModifier {
+    let categories: [BPKImageGalleryCategory<ImageView>]
+    let closeAccessibilityLabel: String
+    let onCloseTapped: () -> Void
+    // @Binding var currentIndex: Int // Category Selected binding?
+    @Binding var isPresented: Bool
+    
+    func body(content: Content) -> some View {
+        content
+            .fullScreenCover(isPresented: $isPresented, content: {
+                ContentView(categories: categories)
+            })
+    }
+    
+    struct ContentView: View {
+        let categories: [BPKImageGalleryCategory<ImageView>]
+        @State var selectedCategory: Int = 0
+        var body: some View {
+            VStack(spacing: .md) {
+                ImageGalleryGridCategoriesCarousel(
+                    categories: categories,
+                    selectedCategory: $selectedCategory
+                )
+                .fixedSize(horizontal: false, vertical: true)
+                
+                ImageGalleryGridImagesGrid(
+                    images: categories[selectedCategory].images
+                )
+            }
+        }
+    }
+}
+
+public extension View {
+    func bpkImageGalleryGrid<Content>(
+        isPresented: Binding<Bool>,
+        categories: [BPKImageGalleryCategory<Content>],
+        closeAccessibilityLabel: String,
+        // currentIndex: Binding<Int>, // Category Selected binding?
+        onCloseTapped: @escaping () -> Void
+    ) -> some View {
+        modifier(
+            ImageGalleryGrid(
+                categories: categories,
+                closeAccessibilityLabel: closeAccessibilityLabel,
+                onCloseTapped: onCloseTapped,
+                isPresented: isPresented
+            )
+        )
+    }
+}
+
+struct ImageGalleryGridCategoriesCarousel<ImageView: View>: View {
+    let categories: [BPKImageGalleryCategory<ImageView>]
+    @Binding var selectedCategory: Int
+    private let categoryImageSideLength: CGFloat = 90
+    
+    var body: some View {
+        ScrollView(.horizontal) {
+            HStack(alignment: .top) {
+                ForEach(0..<categories.count, id: \.self) { ndx in
+                    categoryThumb(for: categories[ndx])
```

Comment:
these may need to be a Button instead for accessibility reasons! (instead of making it tappable)

Reply:
I went for accessibility traits. 
I want to do more of a review of accessibility before this is ready